clients/segments/segments.go

```diff
@@ -113,27 +113,16 @@ func (c Client) GetAll(ctx context.Context) ([]Response, error) {
 	return result, nil
 }
 
-// Upsert creates a new entry of segment on server in case a configuration with an ID doesn't already exist on server. If exists, it updates it.
-// If the method is provided with id = "" creat will be called.
-func (c Client) Upsert(ctx context.Context, id string, data []byte) (Response, error) {
-	//To allow passing empty id for direct call to create without GET
-	existing := &http.Response{StatusCode: http.StatusNotFound}
-	var err error
-	if id != "" {
-		existing, err = c.client.Get(ctx, id, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
-		closeBody(existing)
-		if err != nil {
-			return Response{}, fmt.Errorf(errMsgWithId, "upsert", id, err)
-		}
+// Update puts the content of the segment to the server using http PUT to update segment by id.
+// In the first step GET is called to get mandatory data by the API(owner, uid and version), this is then added to payload.
+func (c Client) Update(ctx context.Context, id string, data []byte) (Response, error) {
+	existing, err := c.client.Get(ctx, id, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	closeBody(existing)
```

Comment:
This won't work. Either defer it or move after you consume the reader of the body

In the test, we have `io.NopCloser(strings.NewReader(apiExistingResource))`.